### PR TITLE
add logdedup processor to contrib

### PR DESCRIPTION
**Pull request by codeboten**
This processor has been marked as alpha in https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/4ddd4e934a1c79c352b95fdcf3979f209a300e07

Pinging release manager for v0.107.0 @dmitryax please merge this change once the contrib v0.107.0 modules are tagged

### CODE REVIEW

**Review by TylerHelmuth — APPROVED** [review]

**Comment by MikeGoldsmith** [comment]
Fingered crossed this can get into the next contrib release 🤞🏻

**Comment by TylerHelmuth** [comment]
@codeboten the k8s distro could take this component as well since it is a vendor-neutral processor.

File: distributions/otelcol-contrib/manifest.yaml [hunk 1]
```diff
@@ -97,6 +97,7 @@ processors:
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/groupbyattrsprocessor v0.106.1
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/groupbytraceprocessor v0.106.1
   - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sattributesprocessor v0.106.1
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/logdedupprocessor v0.106.1
```
Comment (arminru) [inline distributions/otelcol-contrib/manifest.yaml:100]: Note that it did not exist in this version yet as it was only introduced with https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/b92eb433024f91b3e73694571823bf0154f9e1f6 and is first released with v0.107.0, so in this version the build won't pass.

**Review by codeboten — COMMENTED** [review]
Was merged in #636

**Pull request closed without merging**